### ISSUE TITLE
instances should be re-created when image_id changes

### ISSUE THREAD

**Issue by hiscal2015** [terraform-providers/terraform-provider-tencentcloud#78]
provider version: 1.12
when image_id changed, instances should be re-created instead of "modify":

```

  ~ tencentcloud_instance.test_vm_tct_cn[0]
      image_id:                                  "img-mzbr88cf" => "img-axq9idj1"

  ~ tencentcloud_instance.test_vm_tct_cn[1]
      image_id:                                  "img-mzbr88cf" => "img-axq9idj1"

  ~ tencentcloud_instance.test_vm_tct_cn[2]
      image_id:                                  "img-mzbr88cf" => "img-axq9idj1"

  ~ tencentcloud_instance.test_vm_tct_eu[0]
      image_id:                                  "img-o77dd2jn" => "img-pi1pale7"

  ~ tencentcloud_instance.test_vm_tct_eu[1]
      image_id:                                  "img-o77dd2jn" => "img-pi1pale7"

  ~ tencentcloud_instance.test_vm_tct_eu[2]
      image_id:                                  "img-o77dd2jn" => "img-pi1pale7"


Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

tencentcloud_instance.test_vm_tct_cn[0]: Modifying... (ID: ins-1vv9a7er)
  image_id: "img-mzbr88cf" => "img-axq9idj1"
tencentcloud_instance.test_vm_tct_eu[2]: Modifying... (ID: ins-003b1pf2)
  image_id: "img-o77dd2jn" => "img-pi1pale7"
tencentcloud_instance.test_vm_tct_eu[0]: Modifying... (ID: ins-09od6vhg)
  image_id: "img-o77dd2jn" => "img-pi1pale7"
tencentcloud_instance.test_vm_tct_eu[1]: Modifying... (ID: ins-o9wbotcw)
  image_id: "img-o77dd2jn" => "img-pi1pale7"
tencentcloud_instance.test_vm_tct_cn[1]: Modifying... (ID: ins-ac71rjd5)
  image_id: "img-mzbr88cf" => "img-axq9idj1"
tencentcloud_instance.test_vm_tct_cn[2]: Modifying... (ID: ins-7kbqea89)
  image_id: "img-mzbr88cf" => "img-axq9idj1"
```

**Comment by likexian** [comment]
Hello,
We support reset instance's image, re-create is unnecessary.

**Comment by hiscal2015** [comment]
Well, when users updated image_id, they probably want to re-create instances with the new image, why "unnecessary"?

**Comment by likexian** [comment]
Re-create instance will change something, for example ip.
For most user, change image is expect as reinstall the system, but not re-create an instance.

**Comment by hiscal2015** [comment]
> Re-create instance will change something, for example ip.
> For most user, change image is expect as reinstall the system, but not re-create an instance.

I agree with you if user manage instance by UI, but this is terraform, we treat infra as cattle not pets, at least we can have to switch to decide "recreate" or not.

**Comment by ausmartway** [comment]
李总，这个事情上我同意客户的意见，修改了imageid实际上是要求重新创建。

**Comment by likexian** [comment]
> 李总，这个事情上我同意客户的意见，修改了imageid实际上是要求重新创建。
> […]

谢谢刘总，我们修改一下。

**Comment by likexian** [comment]
> > Re-create instance will change something, for example ip.
> > For most user, change image is expect as reinstall the system, but not re-create an instance.
> 
> I agree with you if user manage instance by UI, but this is terraform, we treat infra as cattle not pets, at least we can have a switch to decide "recreate" or not.

Thanks for your advise.

**Comment by hiscal2015** [comment]
@likexian @ausmartway Cool, thanks.

**Comment by likexian** [comment]
#101

**Comment by likexian** [comment]
@hiscal2015 
This has been fixed in v1.15.1, please upgrade and have a try.

**Comment by hiscal2015** [comment]
> @hiscal2015
> This has been fixed in v1.15.1, please upgrade and have a try.

Thanks for fixing this!
